### ISSUE TITLE
Tray icon doesn't show on Fedora 26 and Gnome 3

### ISSUE THREAD

**Issue by phxql** [dorkbox/SystemTray#66]
Hi,

the tray icon doesn't show on Fedora 26 with a Gnome 3 desktop. Output from the log:

```
14:35:51.928 [main] DEBUG dorkbox.systemTray.SystemTray - OS: Linux
14:35:51.928 [main] DEBUG dorkbox.systemTray.SystemTray - Arch: amd64
14:35:51.928 [main] DEBUG dorkbox.systemTray.SystemTray - Oracle Corporation OpenJDK 64-Bit Server VM 1.8.0_144
14:35:51.928 [main] DEBUG dorkbox.systemTray.SystemTray - Is Auto sizing tray/menu? true
14:35:51.928 [main] DEBUG dorkbox.systemTray.SystemTray - Is JavaFX detected? false
14:35:51.929 [main] DEBUG dorkbox.systemTray.SystemTray - Is SWT detected? false
14:35:51.984 [main] DEBUG dorkbox.systemTray.SystemTray - Java Swing L&F: Metal
14:35:51.984 [main] DEBUG dorkbox.systemTray.SystemTray - Auto-detecting tray type
14:35:51.985 [main] DEBUG dorkbox.systemTray.SystemTray - Force GTK2: false
14:35:51.985 [main] DEBUG dorkbox.systemTray.SystemTray - Prefer GTK3: true
14:35:52.006 [main] DEBUG dorkbox.systemTray.SystemTray - Currently using the 'Gnome' desktop environment
14:35:52.006 [main] DEBUG dorkbox.systemTray.SystemTray - Currently using the 'gnome-xorg' session type
14:35:52.054 [main] ERROR dorkbox.systemTray.SystemTray - SystemTray initialization failed. (Unable to discover which implementation to use). Falling back to the Swing Tray.
14:35:52.125 [main] DEBUG dorkbox.util.jna.linux.GtkLoader - GTK: libgtk-3.so.0
14:35:52.125 [GTK Native Event Loop] DEBUG d.util.jna.linux.GtkEventDispatch - Running GTK Native Event Loop
14:35:52.265 [main] DEBUG dorkbox.systemTray.SystemTray - GTK Version: 3.22.21
14:35:52.265 [main] DEBUG dorkbox.systemTray.SystemTray - Is the system already running GTK? false
14:35:52.366 [main] DEBUG dorkbox.systemTray.SystemTray - Tray indicator image size: 22
14:35:52.366 [main] DEBUG dorkbox.systemTray.SystemTray - Tray menu image size: 15
14:35:52.492 [main] DEBUG dorkbox.systemTray.SystemTray - Successfully changed tray icon background color
14:35:52.588 [main] INFO  dorkbox.systemTray.SystemTray - Successfully loaded type: _SwingTray
```

Do i have to install some package?

**Comment by dorkbox** [comment]
Does it show up if you press  `SUPER+M.`?

**Comment by phxql** [comment]
I have installed the TopIconPlus extension. The tray icon isn't visible.

**Comment by dorkbox** [comment]
You might need to install app indicator (not sure what it is called on Fedora, but IIRC on Ubuntu it's libappindicator3).

I haven't tested Fedora 26 yet -- so it might be just a mis-configuration issue. I'll add that to my list of OSes to check.

**Comment by phxql** [comment]
Hi, the package is called `libappindicator`, but installing it doesn't help.

**Comment by tresf** [comment]
> I haven't tested Fedora 26 yet -- so it might be just a mis-configuration issue. I'll add that to my list of OSes to check.

I tested on Fedora 26 + X11 (force-disabled Wayland for better VM mouse/clipboard integration) and it too uses the Swing Java tray.

**Comment by dorkbox** [comment]
This is fixed in Version 3.17, which has just released.
